actors/builtin/miner/miner_actor.go

```diff
@@ -333,6 +337,10 @@ func (a Actor) SubmitWindowedPoSt(rt Runtime, params *SubmitWindowedPoStParams)
 	store := adt.AsStore(rt)
 	var st State
 
+	if !CanWindowPoStProof(params.Proofs[0].PoStProof) {
```

Comment:
params with 0 proofs should be valid up to this point so this can cause a panic. This check should go below the check that there is exactly 1 proof.

Reply:
Thanks for catching that. It revealed a big hole in test coverage for SubmitWindowPost preconditions, which I have now filled in this PR.